Atomic/scan.py

```diff
@@ -34,32 +36,61 @@ def get_scan_info(scanner, scan_type):
                     for x in i['scans']:
                         if x['name'] == scan_type:
                             return i['image_name'], x['args'], i.get('custom_args')
+
+        def set_default_scanner():
+            # If no scanners are defined
+            if len(self.scanners) < 1:
+                count = 1
+                raise ValueError("No scanners are configured for your system.")
+            elif len(self.scanners) == 1:
+                self.default_scanner = self.scanners[0]['scanner_name']
+            else:
+                self.default_scanner = self.atomic_config['default_scanner']
+
         if self.args.debug:
             self.debug = True
 
         # Set debug bool
         self.set_debug()
 
+        set_default_scanner()
+
+        # If a scanner is defined in parseargs
+        if self.args.scanner is not None:
+            self.scanner = self.args.scanner
+        # If only one scanner is defined and no default is defined
+        elif len(self.scanners) == 1 and self.default_scanner is None and self.args.scanner is None:
+            self.scanner = self.scanners[0]['scanner_name']
+        # If more than one scanner is defined but no default
```

Comment:
Should set this default in set_default_scanner, then override it above, eliminating this line altogether.